docs/source/backends.rst

```diff
@@ -108,6 +110,79 @@ supported. An example:
     <COO: shape=(3, 4), dtype=float64, nnz=0, sorted=False, duplicates=False>
 
 
+Autograd
+--------
+
+The `autograd <https://github.com/HIPS/autograd>`_ library is a drop-in for
+``numpy`` that can automatically compute the gradients of array expressions.
+``opt_einsum`` automatically dispatches the ``autograd`` arrays correctly,
+enabling a simple way to compute gradients of tensor contractions:
+
+.. code-block:: python
+
+    >>> import numpy as np
+    >>> import autograd
+    >>> shapes = [(2, 3), (3, 4), (4, 2)]
+    >>> x, y, z = [np.random.rand(*s) for s in shapes]
+
+    >>> # make single arg function as autograd takes derivative of first arg
+    >>> def foo(xyz):
+    ...    return oe.contract('ij,jk,ki->', *xyz)
+
+    >>> foo([x, y, z])
+    array(3.71251519)
+
+    >>> # wrap foo with autograd to compute gradients instead
+    >>> dfoo = autograd.grad(foo)
+    >>> dx, dy, dz = dfoo(arrays)
+    >>> dx, dy, dz
+    (array([[1.30857301, 1.87061563, 0.93695379],
+            [1.0665825 , 1.5318835 , 0.71335157]]),
+     array([[0.52377733, 0.76218852, 0.18797777, 0.60494682],
+            [0.23560851, 0.41939902, 0.05703505, 0.2871758 ],
+            [0.81113451, 1.09167635, 0.32298711, 0.91939656]]),
+     array([[0.5422258 , 1.04966071],
+            [0.29421714, 0.55134473],
+            [0.62781637, 0.93304068],
+            [0.63050881, 0.92410755]]))
+
+Jax
+---
+
+`jax <https://github.com/google/jax>`_ is itself a drop-in for ``autograd``,
+that additionally uses  `XLA <https://www.tensorflow.org/xla>`_ to compile the
+expressions, particularly for the GPU. Using it with ``opt_einsum`` is very
+simple:
+
+.. code-block:: python
+
+    >>> import jax
+    >>> # generate a compiled version of the above function
+    >>> jit_foo = jax.jit(foo)
+    >>> jit_foo([x, y, z])
+    DeviceArray(3.7125154, dtype=float32)
+
+    >>> # generate a compiled version of the gradient function
+    >>> jit_dfoo = jax.jit(jax.grad(foo))
+    >>> jit_dfoo([x, y, z])
+    [DeviceArray([[1.1137383 , 1.14972878, 0.64056885],
```

Comment:
It might be good to use the same random seed above and here to show that they produce the same thing?

Reply:
Ah good catch, I thought I had updated them with the same arrays but obviously forgot. Fixed now.